### PR TITLE
[HTTP/2.0] Extra safety checks to fix crash during test

### PR DESCRIPTION
**Pull request by dheiberg**
Developing tests for the H2 server revealed some bugs. Also notably the server did not catch unexpected exceptions and would not log them.

### CODE REVIEW

File: tools/wptserve/wptserve/server.py [hunk 1]
```diff
@@ -378,8 +378,10 @@ def handle_one_request(self):
                 self.close_connection = True
                 for stream_id, (thread, queue) in stream_queues.items():
                     queue.put(None)
+        except Exception as e:
+            self.logger.debug('(%s) ERROR - Unexpected Error - \n%s' % (self.uid, str(e)))
```
Comment (jgraham) [inline tools/wptserve/wptserve/server.py]: Why isn't this just `self.logger.error` ?

Comment (dhdavvie) [reply tools/wptserve/wptserve/server.py]: oh yes that would make more sense...

Comment (dhdavvie) [reply tools/wptserve/wptserve/server.py]: Updated this, and also changed the other exception catch to `self.logger.error` aswell

File: tools/wptserve/wptserve/server.py [hunk 2]
```diff
@@ -369,17 +369,19 @@ def handle_one_request(self):
                             stream_queues[frame.stream_id] = (self.start_stream_thread(frame, queue), queue)
                         stream_queues[frame.stream_id][1].put(frame)
 
-                        if isinstance(frame, StreamEnded) or frame.stream_ended:
+                        if isinstance(frame, StreamEnded) or (hasattr(frame, "stream_ended") and frame.stream_ended):
                             del stream_queues[frame.stream_id]
 
         except (socket.timeout, socket.error) as e:
-            self.logger.debug('(%s) ERROR - Closing Connection - \n%s' % (self.uid, str(e)))
+            self.logger.error('(%s) ERROR - Closing Connection - \n%s' % (self.uid, str(e)))
```
Comment (jgraham) [inline tools/wptserve/wptserve/server.py]: The formatter will add `ERROR` to the string in this case, so you don't need to.

Comment (dhdavvie) [reply tools/wptserve/wptserve/server.py]: Ahh yes, brain is elsewhere

**Review by jgraham — APPROVED** [review]

**Pull request merged**